src/main/java/com/ait/lienzo/client/core/shape/wires/handlers/impl/WiresDockingControlImpl.java

```diff
@@ -158,20 +174,71 @@ private boolean _isAccept() {
                                                  getShape());
     }
 
-    public static Point2D calculateCandidateLocation(WiresShape shape,
-                                                     WiresContainer parent) {
-        final Point2D location = shape.getControl().getParentPickerControl().getShapeLocation();
-        final Point2D trgAbsOffset = parent.getContainer().getComputedLocation();
-        return new Point2D(location.getX() - trgAbsOffset.getX(),
-                           location.getY() - trgAbsOffset.getY());
+    private Point2D calculateCandidateLocation(WiresShape shape, WiresContainer parent) {
+        return calculateCandidateLocation(shape, getCloserMagnet(shape, parent));
+    }
+
+    private Point2D calculateCandidateLocation(WiresShape shape, WiresMagnet shapeMagnet) {
+        final Point2D location = new Point2D(shapeMagnet.getX(), shapeMagnet.getY());
+        final BoundingBox box = shape.getPath().getBoundingBox();
+        final double newX = location.getX() - (box.getWidth() / 2);
+        final double newY = location.getY() - (box.getHeight() / 2);
+        return new Point2D(newX, newY);
+    }
+
+    private WiresMagnet getCloserMagnet(WiresShape shape,
+                                        WiresContainer parent) {
+        final WiresShape parentShape = (WiresShape) parent;
+        final MagnetManager.Magnets magnets = parentShape.getMagnets();
+        final double shapeWidth = shape.getPath().getBoundingBox().getMaxX();
+        final double shapeHeight = shape.getPath().getBoundingBox().getMaxY();
+
+        int magnetIndex = -1;
+        Double minDistance = null;
+        //not considering the zero magnet, that is the center.
+        for (int i = 1; i < magnets.size(); i++) {
+            WiresMagnet magnet = magnets.getMagnet(i);
+            final Point2D shapeLocation = shape.getComputedLocation();
```

Comment:
calls to `getComputedLocation()` are a bit more "expensive", they consider nested containers and are recursively obtaining the `location` attribute for each parent node, so better moving this outside the loop, makes sense?